### ISSUE TITLE
[Windows] failed to build socket.c under VS2019

### ISSUE THREAD

**Issue by honglei** [zeromq/pyzmq#1394]
```


C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\VC\Tools\MSVC\14.25.28610\bin\HostX86\x64\cl.exe /c /nologo /Ox /W3 /GL /DNDEBUG /MD -IX:\QUIC\libzmq-master\include -Izmq\utils "-IC:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\VC\Tools\MSVC\14.25.28610\ATLMFC\include" "-IC:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\VC\Tools\MSVC\14.25.28610\include" "-IC:\Program Files (x86)\Windows Kits\NETFXSDK\4.6.1\include\um" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\ucrt" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\shared" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\um" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\winrt" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\cppwinrt" -IC:\Python38\include -IC:\Python38\include "-IC:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\VC\Tools\MSVC\14.25.28610\ATLMFC\include" "-IC:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\VC\Tools\MSVC\14.25.28610\include" "-IC:\Program Files (x86)\Windows Kits\NETFXSDK\4.6.1\include\um" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\ucrt" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\shared" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\um" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\winrt" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\cppwinrt" /Tczmq\backend\cython\socket.c /Fobuild\temp.win-amd64-3.8\Release\zmq\backend\cython\socket.obj
socket.c
zmq\backend\cython\socket.c(1501): error C2146: Syntax error: missing ")" (in front of identifier "value")
zmq\backend\cython\socket.c(1501): error C2061: Syntax error: identifier "value"
zmq\backend\cython\socket.c(1501): error C2059: Syntax error: ";"
zmq\backend\cython\socket.c(1501): error C2059: Syntax error: ")"
zmq\backend\cython\socket.c(5502): error C2065: "SOCKET": undeclared identifier
zmq\backend\cython\socket.c(5502): error C2146: Syntax error: missing ";" (in front of identifier "__pyx_v_optval_fd_c")
zmq\backend\cython\socket.c(5502): error C2065: "__pyx_v_optval_fd_c": undeclared identifier
zmq\backend\cython\socket.c(5731): error C2065: "SOCKET": undeclared identifier
zmq\backend\cython\socket.c(5740): error C2065: "__pyx_v_optval_fd_c": undeclared identifier
zmq\backend\cython\socket.c(5751): warning C4013: "__Pyx_PyInt_From_ZMQ_FD_T" is undefined; assume external returns int
zmq\backend\cython\socket.c(5751): error C2065: "__pyx_v_optval_fd_c": undeclared identifier
zmq\backend\cython\socket.c(5751): warning C4047: "=": "PyObject *" and "int" have different levels of indirection
zmq\backend\cython\socket.c(9769): warning C4996:'PyObject_AsWriteBuffer': deprecated in 3.0
zmq\backend\cython\socket.c(9789): warning C4996:'PyObject_AsReadBuffer': deprecated in 3.0
zmq\backend\cython\socket.c(15430): error C2146: Syntax error: missing ")" (in front of identifier "value")
zmq\backend\cython\socket.c(15430): error C2061: Syntax error: identifier "value"
zmq\backend\cython\socket.c(15430): error C2059: Syntax error: ";"
zmq\backend\cython\socket.c(15430): error C2059: Syntax error: ")"
zmq\backend\cython\socket.c(15430): error C2449: found "{" in the file range (are you missing a function header?)
zmq\backend\cython\socket.c(15458): error C2059: Syntax error: "}"
```

**Comment by honglei** [comment]
both pyzmq and libzmq are newest:

libzmq：7c3eaf864cd43d6258ee840a5bf4e17188e78d15
pyzmq ： dd4dac055152d47c829034224cdecf594c7b3f12

**Comment by honglei** [comment]
The related lines 1501/5502/5731:
```c
static CYTHON_INLINE PyObject* __Pyx_PyInt_From_ZMQ_FD_T(ZMQ_FD_T value);
  ZMQ_FD_T __pyx_v_optval_fd_c;
    __pyx_v_sz = (sizeof(ZMQ_FD_T));
```

**Comment by christianaguilera-foundry** [comment]
Caused by https://github.com/zeromq/libzmq/issues/110.

This issue also affected zmqpp: https://github.com/zeromq/zmqpp/issues/233 (https://github.com/zeromq/zmqpp/pull/234)

(Probably affecting also cppzmq.)

**Comment by minrk** [comment]
This should be fixed by ab3f3563d7fe64b9f650ed7481dc2595b2fd28fb
